src/DebugEngineHost/HostDebugger.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.DebugEngineHost
+{
+    /// <summary>
+    /// Provides interactions with the host's debugger
+    /// </summary>
+    public static class HostDebugger
+    {
+        /// <summary>
+        /// Attach to a process using the provided options
+        /// </summary>
+        public static void Attach(string filePath, string options, Guid engineId)
+        {
+            try
+            {
+                ThreadHelper.Generic.BeginInvoke(() => Internal.LaunchDebugTarget(filePath, options, engineId));
```

Comment:
Is it intentional that this is async and returns immediately? Are all the callers prepared for immediate return?